concrete-core-fixture/src/fixture/lwe_ciphertext_discarding_bootstrap.rs

```diff
@@ -0,0 +1,326 @@
+        engine: &mut Engine,
+        context: Self::PreExecutionContext,
+    ) -> Self::PostExecutionContext {
+        let (bootstrap_key, accumulator, mut output_ciphertext, input_ciphertext) = context;
+        unsafe {
+            engine.discard_bootstrap_lwe_ciphertext_unchecked(
+                &mut output_ciphertext,
+                &input_ciphertext,
+                &accumulator,
+                &bootstrap_key,
+            )
+        };
+        (
+            bootstrap_key,
+            accumulator,
+            output_ciphertext,
+            input_ciphertext,
+        )
+    }
+
+    fn process_context(
+        _parameters: &Self::Parameters,
+        maker: &mut Maker,
+        repetition_proto: &Self::RepetitionPrototypes,
+        sample_proto: &Self::SamplePrototypes,
+        context: Self::PostExecutionContext,
+    ) -> Self::Outcome {
+        let (bootstrap_key, accumulator, output_ciphertext, input_ciphertext) = context;
+        let (_, _, proto_glwe_secret_key, _) = repetition_proto;
+        let (proto_plaintext, ..) = sample_proto;
+        let proto_output_ciphertext = maker.unsynthesize_lwe_ciphertext(&output_ciphertext);
+        let proto_output_lwe_secret_key =
+            maker.convert_glwe_secret_key_to_lwe_secret_key(proto_glwe_secret_key);
+        let proto_output_plaintext = <Maker as PrototypesLweCiphertext<
+            Precision,
+            OutputCiphertext::KeyDistribution,
+        >>::decrypt_lwe_ciphertext_to_plaintext(
+            maker,
+            &proto_output_lwe_secret_key,
+            &proto_output_ciphertext,
+        );
+        maker.destroy_lwe_ciphertext(input_ciphertext);
+        maker.destroy_lwe_ciphertext(output_ciphertext);
+        maker.destroy_lwe_bootstrap_key(bootstrap_key);
+        maker.destroy_glwe_ciphertext(accumulator);
+        (
+            maker.transform_plaintext_to_raw(proto_plaintext),
+            maker.transform_plaintext_to_raw(&proto_output_plaintext),
+        )
+    }
+
+    fn compute_criteria(
+        parameters: &Self::Parameters,
+        _maker: &mut Maker,
+        _repetition_proto: &Self::RepetitionPrototypes,
+    ) -> Self::Criteria {
+        let predicted_variance: Variance =
+            fix_estimate_pbs_noise::<Precision::Raw, Variance, OutputCiphertext::KeyDistribution>(
+                parameters.lwe_dimension,
+                parameters.poly_size,
+                parameters.glwe_dimension,
+                parameters.decomp_base_log,
+                parameters.decomp_level_count,
+                parameters.noise,
+            );
+        (predicted_variance,)
+    }
+
+    fn verify(criteria: &Self::Criteria, outputs: &[Self::Outcome]) -> bool {
+        let (means, actual): (Vec<_>, Vec<_>) = outputs.iter().cloned().unzip();
+        assert_delta_std_dev(&actual, means.as_slice(), criteria.0)
+    }
+}
+
+// The current NPE does not use the key distribution markers of concrete-core. This function makes
```

Comment:
Here it would be good to add a FIXME just like for the keyswitch.

Reply:
Done !